[Community Settings]: Flows simplified in all 5 setting sections

micieslak opened the pull request:
### What does the PR do

- SettingsPageHeader control added (generic header intended to be used in all community settings flow.)
- Exclude IntroPanel from WelcomeSettingsView 
	- word wrapping problems fixed
	- storybook page added
- MintedTokensModel refactored to be a regular component instead of a stateful singleton
- MintedTokensSettingsPanel simplified by integrating welcome page into MintedTokensView (Welcome info is a placeholder when list of tokens is empty. For that
reason is hander to keep in MintedTokensView instead of in MintedTokensSettingsPanel and maintain additional state for that.)
- MintTokensSettingsPanel component simplified, storybook page improved
	- management of header simplified, no states needed
	- no need to use StackViewStates helper
	- no need to use internal `d` object as a proxy to pass data between pages
	- StackView used as a base class
	- clipping problems fixed (#11285)
	- scrolling problems fixed (#11289)
	- all flows working from the storybook
	- header management simplified
- SettingsPage component intoduced as a replacement for SettingsPageLayout
- OverviewSettingsPanel refactored 
	- fixed changing position of header when editing community
	- simpler integration with SettingsDirtyToastMessage
- MembersSettingsPanel refactored to use SettingsPage 
	- Minor qml warning fixed
- PermissionsSettingsPanel - welcome view moved to PermissionsView to simplify states management
	- Storybook page updated to cover case when permissions model is empty.
- PermissionsSettingsPanel refactored and simplified, storybook pages fixed and improved
	- header buttions/title/subtitle simplified
	- no need to use states
	- no need to use internal `d` object to pass data between pages
	- permissions list scroll position preserved when navigation back from
	  other views
	- storybook: allow "back" navigation to cover whole flow in sb
	- fix some regressions in permissions-related pages
- AirdropSettingsPanel simplified, storybook page improved 
	- added "back" navigation in storybook
	- no need to use states and StackViewStates component

Closes: #11177
Closes: #11285
Closes: #11298

### Affected areas
Community Settings components

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | bddf9708 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-imports/job/PR-11347/1/) | 2023-06-30 09:22:10 | ~5 min | `tests/imports` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-imports/job/PR-11347/1//consoleText) |

noeliaSD commented on ui/app/AppLayouts/Communities/views/CommunitySettingsView.qml:
```diff
@@ -565,9 +565,6 @@ StatusSectionLayout {
                     target: mintPanel
 
                     function onAirdropToken(tokenKey, type, addresses) {
-                        // Here it is forced a navigation to the new airdrop form, like if it was clicked the header button
```
If this has been removed, probably you also want to remove this `mintPanel.primaryHeaderButtonClicked()` inline **469** ??

micieslak replied:
Fixed!

noeliaSD commented on ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml:
```diff
@@ -257,381 +112,187 @@ SettingsPageLayout {
-                signTransactionPopup.close()
-                burnTokensPopup.close()
-            }
+                    if(preview.isAssetView)
```
Here maybe we can avoid this difference and just have a `mint(token)` signal, since token object contains information about if it's an asset or a collectible indeed.

noeliaSD commented on ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml:
```diff
@@ -257,381 +112,187 @@ SettingsPageLayout {
-                property: "currentToken"
-                value: view.token
-            }
+                        // cleanup dynamically created TokenObject
```
I guess.. leftover

noeliaSD commented on ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml:
```diff
@@ -640,9 +314,10 @@ SettingsPageLayout {
+                        value: tokenViewPage.tokenKey
                     }
                 }
+
```
In line 330 maybe better `readonly property list<Binding> tokenBindings:`

noeliaSD commented on ui/app/AppLayouts/Communities/views/MintedTokensView.qml:
```diff
@@ -52,12 +76,28 @@ StatusScrollView {
         width: root.viewWidth
         spacing: Style.current.halfPadding
 
+        IntroPanel {
```
Maybe we should have a loader here and change source component depending on the case since you will never have the `IntroPanel` with  the rest of components of this `ColumnLayout` shown together. Indeed, the are mutual exclusible.

micieslak replied:
Done

noeliaSD commented on ui/app/AppLayouts/Communities/views/MintedTokensView.qml:
```diff
@@ -52,12 +76,28 @@ StatusScrollView {
             text: qsTr("Assets")
             font.pixelSize: Theme.primaryTextFontSize
             color: Theme.palette.baseColor1
+            visible: root.count !== 0
```
Instead of having this checker in each component below, I guess we can have a component that includes all of them and just show or hide the component on this condition (or as said before, with a loader and change source component).

micieslak replied:
Done

noeliaSD commented on ui/app/AppLayouts/Communities/panels/OverviewSettingsPanel.qml:
```diff
@@ -50,10 +51,13 @@ StackLayout {
     clip: true
 
-    SettingsPageLayout {
+    SettingsPage {
```
Why not using here same approach than with minting or permissions, setting `initialItem`?

micieslak replied:
It could be but now it's as it was so far - StackLayout is used as a base.

noeliaSD reviewed (changes requested):
Refactor looks great! Lots of cleanups and easy to ready components!! 

**Testing issues:**

- All settings sections look incorrect in dark mode:

https://github.com/status-im/status-desktop/assets/97019400/7af9c386-850b-4580-82c0-8cccfbfcf5fd

- Mint token settings: Collectibles / Assets tab name and page title are incorrectly set:

![Screenshot 2023-06-30 at 19 06 51](https://github.com/status-im/status-desktop/assets/97019400/6c35e3aa-496e-4393-8f13-4a7aa98824a6)

- Permissions / Edit page has wrong title, it shoud be `Edit permission` instead of `New permission`:
![Screenshot 2023-06-30 at 19 09 29](https://github.com/status-im/status-desktop/assets/97019400/6533cc34-f5f0-400f-aa2d-7f7d952668c8)

- Maybe you can also fix this warning, seen in storybook when navigating to `Kitty artwork` and state is `deployed`: 
`status-desktop/ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml:397:17: Unable to assign [undefined] to bool`

micieslak commented:
Thank you @noeliaSD for in-depth review!

> * All settings sections look incorrect in dark mode:

Fixed

> * Mint token settings: Collectibles / Assets tab name and page title are incorrectly set:

Fixed

> * Permissions / Edit page has wrong title, it shoud be `Edit permission` instead of `New permission`:

Fixed

> * Maybe you can also fix this warning, seen in storybook when navigating to `Kitty artwork` and state is `deployed`:
>   `status-desktop/ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml:397:17: Unable to assign [undefined] to bool`

Done

caybro reviewed (approved):
Massive! LGTM

noeliaSD reviewed (approved):
Thanks for the fixes!
Tested again, looks great!
